### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by EidHachem**
## In this PR @Ewoskyx and I implemented the below:

- Removed `current_user` method in `ApplicationController`.
- Installed and set up devise.
- Insure the below is working:
   - Can register a new user.
   - User logs in with a combination of email and password.
   - Hashed password should be stored in the database.
   - Ask for confirmation of email.
   - Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class User < ApplicationRecord
-  validates :name, presence: true
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (rloterh) [inline app/models/user.rb:4]: - [ ] As we want (new) users to able to confirm their email kindly ensure that `confirmable` is enabled on device.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<h2>Sign up</h2>
+
```
Comment (rloterh) [inline app/views/devise/registrations/new.html.erb:2]: - [ ] We want your design to closely match the given  login wireframe design. Kindly improve upon your design.


**Your login page:**
image

**Wireframe login page:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Ed, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Devise properly installed

## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (EidHachem) [reply app/models/user.rb:4]: Hello, @rloterh I added :confirmable to devise on line 4 but it is giving me an error after singing in `undefined local variable or method `confirmed_at' ` do you have any suggestion on how to solve it?

**Review by uwadonat — APPROVED** [review]
## Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
good-job-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/user.rb (hunk 1 above)
Comment (rloterh) [reply app/models/user.rb:4]: Hi @EidHachem 
Sorry for the late response, I just noticed your message. However, I believe you've resolved this issue, as I can see from your this Pull Request.

Cheers! 😃

**Pull request merged**